src/Tvheadend.cpp

```diff
@@ -778,7 +778,7 @@ bool CTvheadend::CreateTimer ( const Recording &tvhTmr, PVR_TIMER &tmr )
           tvhTmr.GetDescription().c_str(), sizeof(tmr.strSummary) - 1);
   tmr.state              = tvhTmr.GetState();
   tmr.iPriority          = tvhTmr.GetPriority();
-  tmr.iLifetime          = tvhTmr.GetRetention();
+  tmr.iLifetime          = std::min(tvhTmr.GetRetention(), tvhTmr.GetRemoval()); // Lifetime = the smallest value, retention or removal
```

Comment:
Since this `std::min` logic is now used in four places I would suggest we move it to a separate method, e.g. `Recording::GetLifetime()`.